samples/javascript_nodejs/51.cafe-bot/dialogs/dispatcher/mainDispatcher.js

```diff
@@ -57,6 +61,16 @@ module.exports = {
             this.addDialog(new QnADialog(botConfig, this.userProfileAccessor));
             this.addDialog(new WhoAreYouDialog(botConfig, conversationState, this.userProfileAccessor, onTurnAccessor, this.reservationAccessor));
             this.addDialog(new BookTableDialog(botConfig, this.reservationAccessor, onTurnAccessor, this.userProfileAccessor, conversationState));
+
+            // add recognizer
+            const luisConfig = botConfig.findServiceByNameOrId(LUIS_CONFIGURATION);
+            if (!luisConfig || !luisConfig.appId) throw new Error(`Cafe Dispatch LUIS model not found in .bot file. Please ensure you have all required LUIS models created and available in the .bot file. See readme.md for additional information.\n`);
+            this.luisRecognizer = new LuisRecognizer({
```

Comment:
>LuisRecognizer [](start = 38, length = 14)

In C# side, we housed the recognizers into singleton objects to (in theory) avoid recreation costs on every turn but more to (in theory) maintain the connection..